src/gtnh/assembler/generic_assembler.py

```diff
@@ -0,0 +1,191 @@
+import os
+from pathlib import Path
+from typing import Callable, Dict, List, Optional, Set, Tuple, Union
+from zipfile import ZipFile, ZIP_DEFLATED
+
+from colorama import Fore
+from structlog import get_logger
+
+from gtnh.assembler.downloader import get_asset_version_cache_location
+from gtnh.defs import ModSource, Side
+from gtnh.models.gtnh_config import GTNHConfig
+from gtnh.models.gtnh_release import GTNHRelease
+from gtnh.models.gtnh_version import GTNHVersion
+from gtnh.models.mod_info import ExternalModInfo, GTNHModInfo
+from gtnh.modpack_manager import GTNHModpackManager
+
+log = get_logger(__name__)
+
+
+class GenericAssembler:
```

Comment:
Seems this might be an abstract base class - https://docs.python.org/3/library/abc.html